.github/actions/run-gradle/action.yml

```diff
@@ -86,7 +86,10 @@ runs:
       run: |
         set +e
         ./.github/scripts/thread-dump.sh &
-        timeout ${TIMEOUT_MINUTES}m ./gradlew --build-cache --continue --no-scan \
+        mkdir -p heap-dumps
+        HEAP_DUMP_DIR=$(readlink -f heap-dumps)
+        timeout ${TIMEOUT_MINUTES}m ./gradlew --continue --no-scan \
+        -Dorg.gradle.jvmargs="-Xmx4g -Xss4m -XX:+UseParallelGC -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=$HEAP_DUMP_DIR" \
```

Comment:
maybe we can add comment to `gradle.properties` to remind us to keep the memory configs consistent?